core/arch/arm/plat-zynq7k/plat_init.S

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
+ * All rights reserved.
+ * Copyright (c) 2016, Wind River Systems.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * Entry points for the A9 inits, A9 revision specific or not.
+ * It is assume no stack is available when these routines are called.
+ * It is assume each routine is called with return address in LR
+ * and with ARM registers R0, R1, R2, R3 being scratchable.
+ */
+
+#include <arm32.h>
+#include <arm32_macros.S>
+#include <asm.S>
+#include <kernel/tz_ssvce_def.h>
+#include <kernel/unwind.h>
+#include <platform_config.h>
+
+#define ZYNQ_SLCR_L2C_RAM	0xF8000A1C
+
+.section .text
+.balign 4
+.code 32
+
+/*
+ * void arm_cl2_config(vaddr_t pl310_base) - Memory Cache Level2 Configuration
+ *
+ * Use scratables registers R0-R3.
+ * No stack usage.
+ * LR store return address.
+ * Trap CPU in case of error.
+ */
+FUNC arm_cl2_config , :
+UNWIND(	.fnstart)
+
+	/* disable PL310 ctrl */
+	mov  r1, #0x0
+	str  r1, [r0, #PL310_CTRL]
+
+	/*
+	 * Xilinx AR#54190 recommends setting L2C RAM in SLCR
+	 * to 0x00020202 for proper cache operations.
+	 */
+	ldr  r2, =ZYNQ_SLCR_L2C_RAM
+	ldr  r1, =0x00020202
+	str  r1, [r2]
+
+	/*
+	 * TAG RAM Control Register
+	 *
+	 * bit[10:8]:1 - 2 cycle of write accesses latency
+	 * bit[6:4]:1 - 2 cycle of read accesses latency
+	 * bit[2:0]:1 - 2 cycle of setup latency
+	 */
+	ldr  r2, [r0, #PL310_TAG_RAM_CTRL]
+	movw r1, #0xF888
+	movt r1, #0xFFFF
+	and  r2,r2,r1
+	movw r1, #0xF999
+	movt r1, #0xFFFF
+	orr  r2,r2,r1
+	str  r2, [r0, #PL310_TAG_RAM_CTRL]
```

Comment:
I think I'll keep the L2 cache config code here, in case the bootloader (u-boot) does not set it properly.